Homework_2.playground/Contents.swift

```diff
@@ -0,0 +1,98 @@
+class Stack {
+    private let linkedList = LinkedList()
+    
+    func push(element: Int) {
+        linkedList.append(element)
+    }
+    
+    func pop() -> Int? {
+        return linkedList.removeLast()
+    }
+    
+    func top() -> Int? {
+        return linkedList.last
+    }
+}
+
+class LinkedList {
+    private var head: Frame?
+    var last: Int? {
+        return head?.data
+    }
+    
+    func append(_ element: Int) {
+        head = Frame(data: element, nextFrame: head)
+    }
+    
+    func removeLast() -> Int? {
```

Comment:
See comment above. Technically, you're removing the first element, not the last one.